doc/setup-development-environment.md

```diff
@@ -0,0 +1,167 @@
+`Developer Command Prompt for VS 2017` entry in your "Windows" menu, use that
+entry to create a new shell.
+In that shell install `vcpkg` the Microsoft supported ports for many Open Source
+projects:
+
+```commandline
+c:> cd \Users\%USERNAME%
+c:> git clone --depth 10 https://github.com/Microsoft/vcpkg.git
+c:> cd vcpkg
+c:> .\bootstrap-vcpkg.bat
+c:> .\vcpkg install grpc abseil
+c:> .\vcpkg integrate install
+```
+
+You may need to create a new key pair to connect to GitHub.  Search the web
+for how to do this.  Then you can clone the code:
+
+```commandline
+c:> cd \Users\%USERNAME%\source
+c:> git clone git@github.com:<GITHUB-USERNAME_HERE>/google-cloud-cpp.git
+c:> cd google-cloud-cpp
+c:> git submodule --init --recursive
+```
+
+And compile the code using:
+
+```commandline
+c:> mkdir .build
+c:> cd .build
+c:> cmake -DCMAKE_TOOLCHAIN_FILE=C:/Users/%USERNAME%/vcpkg/scripts/buildsystems/vcpkg.cmake -DGOOGLE_CLOUD_CPP_GRPC_PROVIDER=package ..
+c:> cmake --build . -- /m
+```
+
+Run the tests using:
+
+```commandline
+c:> ctest -C Debug .
+```
+
+### Creating a Windows VM using Google Compute Engine
+
+If you do not have a Windows workstation but need a Windows development
+environment to troubleshoot a test or build problem it might be convenient to
+create a Windows VM.
+The following commands assume you have already created a project:
+
+```commandline
+$ PROJECT_ID=... # Set to your project id
+```
+
+Select a zone to run your VM:
+
+```commandline
+$ gcloud compute zones list
+$ ZONE=... # Pick a zone close to where you are...
+```
+
+Select the name of the VM:
+
+```commandline
+$ VM=... # e.g. VM=my-windows-devbox
+```
+
+Then create the virtual machine using:
+
+```commandline
+$ IMAGE=$(gcloud compute images list --filter="family=('windows-2016')" --sort-by=name | awk '{print $1}' | tail -1)
+$ PROJECT_NUMBER=$(gcloud projects list --filter="project_id=${PROJECT_ID}" | awk '{print $3}' | tail -1)
+$ gcloud compute --project "${PROJECT_ID}" instances create "${VM}" \
+  --zone "${ZONE}" \
+  --image "${IMAGE}" --image-project "windows-cloud" \
+  --boot-disk-size "128" --boot-disk-type "pd-standard" --boot-disk-device-name "${VM}" \
+  --service-account "${PROJECT_NUMBER}-compute@developer.gserviceaccount.com" \
+  --machine-type "n1-standard-8" \
+  --subnet "default" \
+  --maintenance-policy "MIGRATE" \
+  --scopes "https://www.googleapis.com/auth/bigtable.admin","https://www.googleapis.com/auth/bigtable.data","https://www.googleapis.com/auth/cloud-platform"
+```
+
+Reset the password for your account:
+
+```commandline
+$ gcloud compute --project "${PROJECT_ID}" reset-windows-password --zone "${ZONE}" "${VM}"
+```
+
+Save that password in some kind of password manager.  Then connect to the VM
+using your favorite RDP client.  For example the
+[chrome extension](https://chrome.google.com/webstore/detail/chrome-rdp-for-google-clo/mpbbnannobiobpnfblimoapbephgifkm/)
+
+### Connecting to GitHub with PuTTY
```

Comment:
FWIW Windows 10 ships a real OpenSSH client (and server) now. I suspect that will rather quickly become a dominant SSH client for Windows developers.